pkg/client/client.go

```diff
@@ -74,14 +75,37 @@ type Options struct {
 // case of unstructured types, the group, version, and kind will be extracted
 // from the corresponding fields on the object.
 func New(config *rest.Config, options Options) (Client, error) {
-	return newClient(config, options)
+	httpClient, err := rest.HTTPClientFor(config)
+	if err != nil {
+		return nil, err
+	}
+	return newClient(config, httpClient, options)
 }
 
-func newClient(config *rest.Config, options Options) (*client, error) {
+// NewForConfigAndClient returns a new Client using the provided config and Options.
+// The returned client reads *and* writes directly from the server
+// (it doesn't use object caches).  It understands how to work with
+// normal types (both custom resources and aggregated/built-in resources),
+// as well as unstructured types.
+//
+// In the case of normal types, the scheme will be used to look up the
+// corresponding group, version, and kind for the given type.  In the
+// case of unstructured types, the group, version, and kind will be extracted
+// from the corresponding fields on the object.
+// Unlike New, this function allows you to provide your own http.Client.
+func NewForConfigAndClient(config *rest.Config, httpClient *http.Client, options Options) (Client, error) {
+	return newClient(config, httpClient, options)
+}
```

Comment:
Make this private, and modify the `New` only?

Reply:
@vincepri could you expand on what you mean here? Do I leave the New function as-is or do I add the `httpClient *http.Client` to the New function?

Reply:
httpClient can be just an `Options` and default to what `rest.HTTPClientFor` returns